tiledb/sm/filesystem/s3.cc

```diff
@@ -947,40 +976,53 @@ Status S3::init_client() const {
   }
 #endif
 
+  if (!aws_access_key_id.empty() && !aws_secret_access_key.empty() && !aws_role_arn.empty()){
+    return Status::S3Error("Authentication credentials can exclusively be either long lived access credentials"
```

Comment:
Maybe "Ambiguous authentication credentials; both permanent and temporary authentication credentials are configured"